pkg/checks/checks.go

```diff
@@ -0,0 +1,81 @@
+package checks
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// WaitForHTTPStatusOk poll until URL returns 200
+func WaitForHTTPStatusOk(transport http.RoundTripper, urltocheck string) error {
+	cli := &http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	req, err := http.NewRequest("GET", urltocheck, nil)
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := cli.Do(req)
+		if err, ok := err.(*url.Error); ok && (err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF) {
+			time.Sleep(time.Second)
+			continue
+		}
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+
+		time.Sleep(time.Second)
+	}
+}
+
+// CheckStatusOfPods verifies all pods are in running state
+func CheckStatusOfPods(coreclient *coreclient.CoreV1Client, namespace string) error {
```

Comment:
use a context here and in WaitForHTTPStatusOk and return if the context has expired.

Caller:
```
ctx, cancel := context.WithTimeout(ctx.Background(), 5 * time.Minute)
defer cancel()
f(ctx, ...)
```

Callee:
```
func f(ctx context.Context, ...) {
	...
	for {
		...
		select {
		case <-ctx.Done():
			return ctx.Err()
		default:
		}
	}
}
```

Reply:
(then no need for startTime, hard-coded timeout, etc)

Reply:
How about `func WaitForPodRunning(coreclient *coreclient.CoreV1Client, namespace, name string)` and have a list of pods you want to check for now, a bit like in https://github.com/Azure/acs-engine/blob/master/test/e2e/openshift/openshift_test.go#L145-L173